### PR TITLE
feature(on-ramp): update supported networks

### PR DESCRIPTION
**Pull request by wachunei**
## Explanation

This PR updates the supported networks in the portfolio "buy crypto" (on-ramp) feature. It changes every link to the feature to have chainId, selectedAddress and metametricsId as search params.

The added networks are: Cronos, Moonbeam, Moonriver, Aurora, Harmony and Palm.

## Screenshots/Screencaps

_N/A_

### Before

_N/A_
### After

_N/A_
## Manual Testing Steps

Every link to "Buy" or "Buy <ASSET>" should now contain the query parameters and be displayed in the new supported networks.

## Pre-merge author checklist

- [x] I've clearly explained:
  - [x] What problem this PR is solving
  - [x] How this problem was solved
  - [x] How reviewers can test my changes
- [x] Sufficient automated test coverage has been added

## Pre-merge reviewer checklist

- [ ] Manual testing (e.g. pull and build branch, run in browser, test code being changed)
- [ ] PR is linked to the appropriate GitHub issue
- [ ] **IF** this PR fixes a bug in the release milestone, add this PR to the release milestone

If further QA is required (e.g. new feature, complex testing steps, large refactor), add the `Extension QA Board` label.

In this case, a QA Engineer approval will be be required.

### CODE REVIEW

File: ui/hooks/experiences/useRamps.ts [hunk 1]
```diff
@@ -13,13 +18,23 @@ const portfolioUrl = process.env.PORTFOLIO_URL;
...
   const getBuyURI = useCallback((_chainId: Hex) => {
     switch (_chainId) {
       case CHAIN_IDS.SEPOLIA:
         return 'https://faucet.sepolia.dev/';
-      default:
-        return `${portfolioUrl}/buy?metamaskEntry=ext_buy_button`;
+      default: {
+        const params = new URLSearchParams();
+        params.set('metamaskEntry', 'ext_buy_button');
+        params.set('address', selectedAddress);
```
Comment (Gudahtt) [inline ui/hooks/experiences/useRamps.ts:31]: I noticed that this wasn't included previously. Is this change related to adding networks? It's not clear why this is now needed

Comment (wachunei) [reply ui/hooks/experiences/useRamps.ts:31]: We used this pattern with the [previous on-ramp with all providers](https://github.com/MetaMask/metamask-extension/blob/eee4cc3100f02282e7e008b92d4a503284aa48ce/app/scripts/lib/buy-url.js#L68) and we are bringing it back now. We've seen Portfolio conversion data showed a significant drop-off for Extension users due to the connect wallet friction.

Comment (Gudahtt) [reply ui/hooks/experiences/useRamps.ts:31]: Perhaps we could consider this addition in a separate PR? I'm aware of that problem, but it seems distinct from the addition of new supported networks, and there are other solutions under consideration.

**Review by Gudahtt — APPROVED** [review]
LGTM!

**Review by brad-decker — APPROVED** [review]
